sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/AadCredentialBuilderBase.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+/**
+ * The base class for credential builders that allow specifying a client id and tenant ID for an Azure Active Directory.
+ * @param <T> the type of the credential builder
+ */
+public abstract class AadCredentialBuilderBase<T extends AadCredentialBuilderBase<T>> extends CredentialBuilderBase<T> {
+    String clientId;
+    String tenantId;
+
+    /**
+     * Sets the client ID of the application.
+     * @param clientId the client ID of the application.
+     * @return {@link <T>} itself
+     */
+    @SuppressWarnings("unchecked")
+    public T clientId(String clientId) {
+        this.clientId = clientId;
+        return (T) this;
+    }
+
+    /**
+     * Sets the tenant ID of the application.
+     * @param tenantId the tenant ID of the application.
+     * @return {@link <T>} itself
+     */
+    @SuppressWarnings("unchecked")
+    public T tenantId(String tenantId) {
+        this.tenantId = tenantId;
+        return (T) this;
+    }
```

Comment:
tenantId isn't needed for some of the buliders deriving from this type such as InteractiveBrowserCredentialBuilder and  DeviceCodeCredentialBuilder

Reply:
Moved to ClientSecretCredentialBuilder and ClientCertificateCredentialBuilder